lib/chore/queues/sqs/publisher.rb

```diff
@@ -3,51 +3,64 @@
 module Chore
   module Queues
     module SQS
-      
       # SQS Publisher, for writing messages to SQS from Chore
       class Publisher < Chore::Publisher
         @@reset_next = true
 
+        # @param [Hash] opts Publisher options
         def initialize(opts={})
           super
           @sqs_queues = {}
           @sqs_queue_urls = {}
         end
 
-        # Takes a given Chore::Job instance +job+, and publishes it by looking up the +queue_name+.
+        # Publishes a message to an SQS queue
+        #
+        # @param [String] queue_name Name of the SQS queue
+        # @param [Hash] job Job instance definition, will be encoded to JSON
+        #
+        # @return [struct Aws::SQS::Types::SendMessageResult]
         def publish(queue_name,job)
-          queue = self.queue(queue_name)
-          queue.send_message(encode_job(job))
+          queue = queue(queue_name)
+          queue.send_message(message_body: encode_job(job))
         end
 
         # Sets a flag that instructs the publisher to reset the connection the next time it's used
+        # TODO: We reset on every publish. Is this desirable? Seems like a performance problem.
+        #
+        # @return [Array<Seahorse::Client::NetHttp::ConnectionPool>]
         def self.reset_connection!
           @@reset_next = true
+          Aws.empty_connection_pools!
```

Comment:
Just noting that this is also impacted by the same comment on where the logic should live